lib/eslint/eslint.js

```diff
@@ -314,7 +358,33 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath) && hasUnstableTSConfigFlag;
+
+    const isBun = isRunningInBun();
+
+    const isDeno = isRunningInDeno();
+
+    if (isTS && !isDeno && !isBun) {
+
+        const createJiti = await import("jiti").then(jitiModule => jitiModule?.createJiti ?? jitiModule.default, () => {
```

Comment:
A reminder that we should remove the ` ?? jitiModule.default` fallback when jiti 2 is released. We should also update the `devDependencies` and `peerDependencies` versions in package.json to match only jiti v2 stable.